presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPickTableLayout.java

```diff
@@ -101,6 +102,20 @@ public void doesNotFireIfTableScanHasConstraint()
                 .doesNotFire();
     }
 
+    @Test
+    public void doesNotFireIfTableScanNothingUpdated()
```

Comment:
This test passes on current `master` too 😕

Reply:
It is because there was a bug, one of the firsts commit fixes that. Rule is not fired when either pattern do not match or apply return empty(). Previously pattern did not match as `originalConstraint == BoolLiteral.TRUE` did not pass even when originalConstraint was `true`.